TestResumedFetch failed on debian-8 (flavor coreos)

**Environment**

Replace this with the output of:
`printf "$(rkt version)\n--\n$(uname -srm)\n--\n$(cat /etc/os-release)\n--\n$(systemctl --version)"`

**What did you do?**

- Run the tests in Jenkins with debian-8, coreos flavor.

**What did you expect to see?**

TestResumedFetch passes fine.

**What did you see instead?**

```
06:48:43 === RUN   TestFetchFullHash
06:48:43 --- PASS: TestFetchFullHash (0.62s)
06:48:43 === RUN   TestResumedFetch
06:48:43 2016/05/20 06:48:43 http: panic serving 127.0.0.1:62908: send on closed channel
06:48:43 goroutine 174 [running]:
06:48:43 net/http.(*conn).serve.func1(0xc8204e0700)
06:48:43 	/usr/lib/go/src/net/http/server.go:1389 +0xc1
06:48:43 panic(0x9b2620, 0xc8201ec9a0)
06:48:43 	/usr/lib/go/src/runtime/panic.go:426 +0x4e9
06:48:43 github.com/coreos/rkt/tests.testInterruptingServerHandler.func1(0x7f6083bbf0d0, 0xc8203b8410, 0xc820098540)
06:48:43 	/home/admin/workspace/rkt-matrix/flavor/coreos/label/debian-8/builds/build-rkt-coreos/build-rkt-1.6.0+git/gopath/src/github.com/coreos/rkt/tests/rkt_fetch_test.go:388 +0xced
06:48:43 github.com/coreos/rkt/tests.testServerHandler.func1(0x7f6083bbf0d0, 0xc8203b8410, 0xc820098540)
06:48:43 	/home/admin/workspace/rkt-matrix/flavor/coreos/label/debian-8/builds/build-rkt-coreos/build-rkt-1.6.0+git/gopath/src/github.com/coreos/rkt/tests/rkt_fetch_test.go:313 +0x63
06:48:43 net/http.HandlerFunc.ServeHTTP(0xc8203d8240, 0x7f6083bbf0d0, 0xc8203b8410, 0xc820098540)
06:48:43 	/usr/lib/go/src/net/http/server.go:1618 +0x3a
06:48:43 net/http.serverHandler.ServeHTTP(0xc8204e0100, 0x7f6083bbf0d0, 0xc8203b8410, 0xc820098540)
06:48:43 	/usr/lib/go/src/net/http/server.go:2081 +0x19e
06:48:43 net/http.(*conn).serve(0xc8204e0700)
06:48:43 	/usr/lib/go/src/net/http/server.go:1472 +0xf2e
06:48:43 created by net/http.(*Server).Serve
06:48:43 	/usr/lib/go/src/net/http/server.go:2137 +0x44e
06:48:43 --- FAIL: TestResumedFetch (0.32s)
06:48:43 	rkt_tests.go:115: Running command: /home/admin/workspace/rkt-matrix/flavor/coreos/label/debian-8/builds/build-rkt-coreos/build-rkt-1.6.0+git/tmp/functional/rkt --dir=/tmp/datadir-696351272 --local-config=/tmp/localdir-240357223 --system-config=/tmp/systemdir-956429210 --no-store --insecure-options=image fetch http://127.0.0.1:17976
06:48:43 	rkt_tests.go:115: Running command: /home/admin/workspace/rkt-matrix/flavor/coreos/label/debian-8/builds/build-rkt-coreos/build-rkt-1.6.0+git/tmp/functional/rkt --dir=/tmp/datadir-696351272 --local-config=/tmp/localdir-240357223 --system-config=/tmp/systemdir-956429210 --no-store --insecure-options=image fetch http://127.0.0.1:17976
06:48:43 	rkt_fetch_test.go:263: hash didn't match: ExpectRegex didn't find regex '.*sha512-9c2dd424b6cbb964e9cc3ddae6ec1dbe'.
```

--- a/tests/rkt_fetch_test.go
+++ b/tests/rkt_fetch_test.go
@@ -28,6 +28,7 @@ import (
 	"strings"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/appc/spec/schema/types"
 	"github.com/coreos/rkt/tests/testutils"
@@ -395,6 +396,9 @@ func testInterruptingServerHandler(t *testing.T, imagePath string, kill, waitfor
 			panic(err)
 		}
 
+		// sleep a bit before signaling that rkt should be killed since it
+		// might not have had time to write everything to disk
+		time.Sleep(time.Second)
 		kill <- struct{}{}
 		<-waitforkill
 	}

--- a/tests/rkt_fetch_test.go
+++ b/tests/rkt_fetch_test.go
@@ -165,7 +165,7 @@ func testFetchDefault(t *testing.T, arg string, image string, imageArgs string, 
 	err := child.Wait()
 	status := getExitStatus(err)
 	if status != 0 {
-		t.Errorf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, 0, child.Collect())
+		t.Logf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, 0, child.Collect())
 		t.Skip("remote fetching failed, probably a network failure. Skipping...")
 	}
 
@@ -212,7 +212,7 @@ func testFetchNoStore(t *testing.T, args string, image string, imageArgs string,
 	err := child.Wait()
 	status := getExitStatus(err)
 	if status != 0 {
-		t.Errorf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, 0, child.Collect())
+		t.Logf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, 0, child.Collect())
 		t.Skip("remote fetching failed, probably a network failure. Skipping...")
 	}
 }

